src/workerd/jsg/jsvalue.h

```diff
@@ -426,15 +427,18 @@ inline kj::Maybe<T&> JsValue::tryGetExternal(Lock& js, const JsValue& value) {
 }
 
 template <typename T>
+  requires(std::same_as<T, uint16_t> || std::same_as<T, kj::byte>)
 inline kj::Array<T> JsString::toArray(Lock& js, WriteOptions options) const {
   if constexpr (kj::isSameType<T, kj::byte>()) {
     KJ_ASSERT(inner->ContainsOnlyOneByte());
     auto buf = kj::heapArray<kj::byte>(inner->Length());
     inner->WriteOneByte(js.v8Isolate, buf.begin(), 0, buf.size(), options);
     return kj::mv(buf);
   } else {
-    auto buf = kj::heapArray<uint16_t>(inner->Length());
-    inner->Write(js.v8Isolate, buf.begin(), 0, buf.size(), options);
+    v8::String::ValueView view(js.v8Isolate, inner);
+    KJ_ASSERT(!view.is_one_byte());
```

Comment:
This also changes behavior -- previously, if you had a one-byte string and asked to create an `Array<uint16_t>`, this worked fine, it would just expand the characters to two-byte.

I also think the change here probably doesn't improve performance at all as `v8::String::Write()` probably does the same thing internally.

Reply:
I changed it to support one byte to two-byte conversion, but v8::String::Write does more than just memcpy, since it has to flatten the underlying string before making any operations at all.